apps/docs/src/pages/en/features/completion.md

```diff
@@ -0,0 +1,162 @@
+---
+title: CompletionFactory
+layout: ../../../layouts/MainLayout.astro
+---
+
+So for now, we've got a CLI with all the needed tools. The problem is - the end
+user does not know what commands are available. We can fix that with the
+`CompletionFactory`! This service has the ability to generate a completion
+script for the end user to use, both for ZSH & Bash and Fig! The completion will
+be generated dynamically based on the commands and options available.
+
+# Setup Your Completion Using The CompletionFactory
+
+Your `main.ts` file:
+
+```typescript
+import { CommandFactory, CompletionFactory } from 'nest-commander';
+import { AppModule } from './app.module';
+
+(async () => {
+  const app = await CommandFactory.createWithoutRunning(AppModule);
+
+  const cliCommand = 'YOUR-CLI-COMMAND';
+  const cliExecutable = 'YOUR-CLI-EXECUTABLE';
+
+  CompletionFactory.registerCompletionCommand(app, {
+    cmd: cliCommand,
+    fig: true,
+    nativeShell: {
+      executablePath: cliExecutable
+    }
+  });
+
+  await CommandFactory.runApplication(app);
+})();
+```
+
+<br>
```

Comment:
I noticed a couple of `<br>` and `<hr>` elements. Any reason for them over regular markdown?

Reply:
Just for a better looking preview.